echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/GitEventMonitor.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.pipelinetriggers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spectator.api.Registry;
+import com.netflix.spinnaker.echo.events.EchoEventListener;
+import com.netflix.spinnaker.echo.model.*;
+import lombok.NonNull;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import rx.Observable;
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+/**
+ * Triggers pipelines on _Orca_ when a trigger-enabled build completes successfully.
+ */
+@Component
+@Slf4j
+public class GitEventMonitor implements EchoEventListener {
+
+  public static final String GIT_TRIGGER_TYPE = "git";
+
+  private final ObjectMapper objectMapper = new ObjectMapper();
+
+  private final PipelineCache pipelineCache;
+  private final Action1<Pipeline> subscriber;
+  private final Registry registry;
+
+  @Autowired
+  public GitEventMonitor(@NonNull PipelineCache pipelineCache,
+                         @NonNull Action1<Pipeline> subscriber,
+                         @NonNull Registry registry) {
+    this.pipelineCache = pipelineCache;
+    this.subscriber = subscriber;
+    this.registry = registry;
+  }
+
+  @Override
+  public void processEvent(Event event) {
+    if (!event.getDetails().getType().equalsIgnoreCase(GitEvent.TYPE)) {
+      return;
+    }
+
+    GitEvent buildEvent = objectMapper.convertValue(event, GitEvent.class);
+    Observable.just(buildEvent)
+      .doOnNext(this::onEchoResponse)
+      .subscribe(triggerEachMatchFrom(pipelineCache.getPipelines()));
+  }
+
+  private void onEchoResponse(final GitEvent event) {
+    registry.gauge("echo.events.per.poll", 1);
+  }
+
+  private Action1<GitEvent> triggerEachMatchFrom(final List<Pipeline> pipelines) {
+    return event -> {
+      Observable.from(pipelines)
```

Comment:
otherwise we would need to map every possible field from orca

Would suggest adding a TriggerEvent class for orca-pipelinetriggers and model Git, Cron and BuildEvents off that since it is essentially a different module and we don't have to deal with orca events as triggers.